deep_quoridor/src/utils/timer.py

```diff
@@ -0,0 +1,83 @@
+import time
+from typing import Optional
+
+import wandb
+import wandb.wandb_run
+
+from utils.misc import format_time
+
+
+class Timer:
+    """
+    Class to measure elapsed time and log it to the console and optionally wandb.
+    It's a class just to create a container for the functions and variables, it can't be instantiated.
+    TO DO: would be nice if it would easily support multi-process, so we can gather information coming
+    from different processes and compile it.
+    """
+
+    starts = {}
+    counts = {}
+    totals = {}
+    wandb_run: Optional[wandb.wandb_run.Run] = None
+
+    def __new__(cls, *args, **kwargs):
+        raise TypeError("Timer is a static class and cannot be instantiated.")
+
+    @classmethod
+    def set_wandb_run(cls, wandb_run: wandb.wandb_run.Run):
+        cls.wandb_run = wandb_run
+
+    @classmethod
+    def start(cls, name: str):
+        if name in cls.starts:
+            print(f"TIMER: WARNING - timer for {name} was already started")
+
+        cls.starts[name] = time.perf_counter()
+
+    @classmethod
```

Comment:
Having these be class methods makes some things easier - we'll just need to be careful if we ever add multi-threading.